src/cascadia/TerminalSettings/KeyChord.cpp

```diff
@@ -39,4 +115,171 @@ namespace winrt::Microsoft::Terminal::Settings::implementation
     {
         _vkey = value;
     }
+
+    // Method Description:
+    // - Deserializes the given string into a new KeyChord instance. If this
+    //   fails to translate the string into a keychord, it will throw a
+    //   hresult_invalid_argument exception.
+    // - The string should fit the format "[ctrl+][alt+][shift+]<keyName>",
+    //   where each modifier is optional, and keyName is either one of the
+    //   names listed in the vkeyNamePairs vector above, or is one of 0-9a-zA-Z.
+    // Arguments:
+    // - hstr: the string to parse into a keychord.
+    // Return Value:
+    // - a newly constructed KeyChord
+    winrt::Microsoft::Terminal::Settings::KeyChord KeyChord::FromString(const hstring& hstr)
+    {
+        std::wstring wstr{ hstr };
+
+        // Split the string on '+'
+        std::wstring temp;
+        std::vector<std::wstring> parts;
+        std::wstringstream wss(wstr);
+
+        while(std::getline(wss, temp, L'+'))
+        {
+            parts.push_back(temp);
+        }
+
+        // If we have > 4, something's wrong.
+        if (parts.size() > 4)
```

Comment:
4 should be a named constant

Reply:
Also throw if `parts.size() == 1`. I don't think we want those to go through.

---
In reply to: [283562006](https://github.com/microsoft/Terminal/pull/748#discussion_r283562006) [](ancestors = 283562006)

Reply:
@carlos-zamora I mean, if someone _really_ wants `t` to open a new tab, who are we to stop them?